### PR TITLE
Add support for WebP

### PR DESCRIPTION
**Pull request by jbrooksuk**
Replaces https://github.com/thephpleague/glide/pull/202

### CODE REVIEW

**Comment by jbrooksuk** [comment]
Travis doesn't have WebP support enabled for PHP 5.4 https://github.com/travis-ci/travis-ci/issues/8435

**Comment by reinink** [comment]
> Travis doesn't have WebP support enabled for PHP 5.4

Meaning it's supported in later versions of PHP? Honestly, it's probably getting to the point that we could drop support for PHP 5 entirely.

**Comment by jbrooksuk** [comment]
>> Travis doesn't have WebP support enabled for PHP 5.4 ...

@reinink as in, it PHP 5.4 does support it, but Travis hasn’t built PHP to include the extension required. I’d agree though, 5.* is coming to complete EOL. http://php.net/supported-versions.php

**Comment by reinink** [comment]
Yeah 10-4. It looks like all the PHP 5.X builds are failing:

image

**Comment by jbrooksuk** [comment]
👍 

Thoughts on dropping PHP 5 support in a separate PR?

**Comment by reinink** [comment]
I think I'm going to keep 5.X support in 1.X. I'd like to find a way to make Travis ignore these tests on 5.X somehow.

I know it's hard to believe (), but I'm planning on getting back into my 2.0 release soon, and I think we'll make that a 7.X only version.

File: src/Manipulators/Encode.php [hunk 1]
```diff
@@ -46,6 +46,7 @@ public function getFormat(Image $image)
             'jpg' => 'image/jpeg',
             'pjpg' => 'image/jpeg',
             'png' => 'image/png',
+            'webp' => 'image/webp',
```
Comment (jbrooksuk) [inline src/Manipulators/Encode.php]: @reinink we could check for the existence of the [`imagewebp`](http://www.php.net/manual/en/function.imagewebp.php) function before adding this format.

Comment (reinink) [reply src/Manipulators/Encode.php]: Yes! I like that. Can you add that?

Comment (jbrooksuk) [reply src/Manipulators/Encode.php]: I've added the condition, but not sure if it'll work.

Comment (jbrooksuk) [reply src/Manipulators/Encode.php]: I had to add it elsewhere 👍

**Comment by jbrooksuk** [comment]
> I think I'm going to keep 5.X support in 1.X. I'd like to find a way to make Travis ignore these tests on 5.X somehow.

Done!

> I know it's hard to believe (), but I'm planning on getting back into my 2.0 release soon, and I think we'll make that a 7.X only version.

That'd be great. I'd love to be able to add custom commands (like for instance, a soft blur).

**Comment by reinink** [comment]
Tagged: https://github.com/thephpleague/glide/releases/tag/1.4.0

**Pull request merged**